src/rgw/rgw_rest_s3.cc

```diff
@@ -1713,8 +1713,19 @@ int RGWPostObj_ObjStore_S3::get_policy()
 
     op_ret = rgw_get_user_info_by_access_key(store, s3_access_key, user_info);
     if (op_ret < 0) {
+      class LDAPTokenExtractor : public RGWTokenBasedAuthEngine::Extractor {
```

Comment:
Well, I have defined it inline, because the class will just be used by this method. Like I said I will improve it later.